src/Reflection/ReflectionNamedType.php

```diff
@@ -54,6 +61,48 @@ public function isBuiltin(): bool
         return array_key_exists(strtolower($this->name), self::BUILT_IN_TYPES);
     }
 
+    public function getClass(): ReflectionClass
+    {
+        if (! $this->isBuiltin()) {
+            return $this->reflector->reflectClass($this->name);
+        }
+
+        if (
+            $this->owner instanceof ReflectionEnum
+            || $this->owner instanceof ReflectionFunction
+            || ($this->owner instanceof ReflectionParameter && $this->owner->getDeclaringFunction() instanceof ReflectionFunction)
+        ) {
+            throw new LogicException('This type does not have class type');
```

Comment:
We should add `$this->name` to the exception

Reply:
Modified to:

```throw new LogicException(sprintf('The type %s cannot be resolved to class', $this->name));```